### PR TITLE
[IMP] mail: adapt component tests data seeding

### PR DESCRIPTION
**Pull request by tsm-odoo**

### CODE REVIEW

File: addons/mail/static/tests/qunit_suite_tests/components/discuss_sidebar_category_tests.js [hunk 1]
```diff
@@ -240,12 +247,13 @@ QUnit.test('channel - states: open manually by clicking the title', async functi
...
-    this.data['res.users.settings'].records.push({
-        user_id: this.data.currentUserId,
+    const pyEnv = await startServer();
+    pyEnv['mail.channel'].create();
+    pyEnv['res.users.settings'].create({
+        user_id: pyEnv.currentUserId,
         is_discuss_sidebar_category_channel_open: true,
     });
-    const currentUserId = this.data.currentUserId;
+    const currentUserId = pyEnv.currentUserId;
```
Comment (phenix-factory) [inline addons/mail/static/tests/qunit_suite_tests/components/discuss_sidebar_category_tests.js:256]: This `const` could be removed

File: addons/mail/static/tests/qunit_suite_tests/components/message_tests.js [hunk 2]
```diff
@@ -812,14 +796,12 @@ QUnit.test('data-oe-id & data-oe-model link redirection on click', async functio
...
-        data: this.data,
-        hasChatWindow: true,
-    });
+    const pyEnv = await startServer();
+    const resPartnerId1 = pyEnv['res.partner'].create();
+    pyEnv['res.users'].create({ partner_id: resPartnerId1 });
+    const { createMessageComponent, messaging } = await start({ hasChatWindow: true });
     const message = messaging.models['Message'].create({
-        author: insert({ id: 10 }),
+        author: insert({ id: resPartnerId1 }),
```
Comment (phenix-factory) [inline addons/mail/static/tests/qunit_suite_tests/components/message_tests.js:804]: Should we can remove this `create` from messaging now ?

Comment (phenix-factory) [reply addons/mail/static/tests/qunit_suite_tests/components/message_tests.js:804]: There is multiple occurrences of this in this file.

**Review by phenix-factory — APPROVED** [review]
Can't find anything meaningful and the tests are green.
LGTM

**Review by seb-odoo — APPROVED** [review]
@robodoo r+

File: addons/mail/static/tests/qunit_suite_tests/components/message_tests.js (hunk 2 above)
Comment (seb-odoo) [reply addons/mail/static/tests/qunit_suite_tests/components/message_tests.js:804]: Yes but this is complicated, naive tests needs to be adapted to actually fetch data from server

**Pull request closed without merging**